landing-strip/core/src/main/java/com/novoda/landingstrip/TabsState.java

```diff
@@ -0,0 +1,189 @@
+package com.novoda.landingstrip;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.v4.view.PagerAdapter;
+import android.support.v4.view.ViewPager;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import com.novoda.landingstrip.LandingStrip.TabSetterUpper;
+
+public class TabsState implements OnPagerAdapterChangedListener {
```

Comment:
:+1: Yes, I understand your concerns. I'm struggling to reduce it down, though. Possibly that is a smell in itself!

If I can break it up a little, we remove the need for the `Builder` which solves that problem, as well.

Leave it with me...